From c5d391adba7bd024954d0df1e0405f5749f4d4ca Mon Sep 17 00:00:00 2001
From: Dudemanguy <random342@airmail.cc>
Date: Tue, 18 Aug 2026 10:27:43 -0500
Subject: [PATCH] ao_coreaudio: register hotplug after succesful audiounit init

No real point in registering the hotplug if we don't even know if the
actual ao is going to work or not yet. Just move it down.
---
 audio/out/ao_coreaudio.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/audio/out/ao_coreaudio.c b/audio/out/ao_coreaudio.c
index 4a582bbc210d872818837422cc2b1012b2bdfab9..7d421390e9dc3730a331c175d8714920d0d36fa3 100644
--- a/audio/out/ao_coreaudio.c
+++ b/audio/out/ao_coreaudio.c
@@ -168,9 +168,6 @@ static int init(struct ao *ao)
     if (!reinit_device(ao))
         goto coreaudio_error;
 
-    if (!register_hotplug_cb(ao))
-        goto coreaudio_error;
-
     if (p->change_physical_format)
         init_physical_format(ao);
 
@@ -188,6 +185,9 @@ static int init(struct ao *ao)
     if (!r)
         goto coreaudio_error;
 
+    if (!register_hotplug_cb(ao))
+        goto coreaudio_error;
+
     reinit_latency(ao);
     ao->device_buffer = av_rescale(p->hw_latency_ns, ao->samplerate, 1000000000) * 2;
 
